tiled/server/router.py

```diff
@@ -1352,6 +1354,76 @@ async def put_awkward_full(
     return json_or_msgpack(request, None)
 
 
+@router.patch("/metadata/{path:path}", response_model=schemas.PatchMetadataResponse)
+async def patch_metadata(
+    request: Request,
+    body: schemas.PatchMetadataRequest,
+    validation_registry=Depends(get_validation_registry),
+    settings: BaseSettings = Depends(get_settings),
+    entry=SecureEntry(scopes=["write:metadata"]),
+):
+    if not hasattr(entry, "patch_metadata"):
+        raise HTTPException(
+            status_code=405, detail="This node does not support update of metadata."
+        )
+
+    if request.headers["content-type"] == "application/json-patch+json":
+        metadata = apply_json_patch(
+            entry.metadata(), body.patch if body.patch is not None else []
+        )
+    elif request.headers["content-type"] == "application/merge-patch+json":
+        metadata = apply_merge_patch(
+            entry.metadata(), body.patch if body.patch is not None else {}
+        )
+    else:
+        raise HTTPException(
+            status_code=406,
+            detail="application/json-patch+json or application/merge-patch+json expected.",
+        )
+
+    structure_family, structure, specs = (
+        entry.structure_family,
+        entry.structure(),
+        body.specs if body.specs is not None else entry.specs,
+    )
+
+    metadata_modified = False
+
+    # Specs should be ordered from most specific/constrained to least.
```

Comment:
Can this block be factored out into a utility function instead of duplicated?

Reply:
Will do. I think that is a sensible approach.